packages/design/src/Form/components/GenderId/index.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
+import { useFormContext } from 'react-hook-form';
+import { type GenderIdProps } from '@atj/forms';
+
+import { type PatternComponent } from '../../index.js';
+
+const GenderIdPattern: PatternComponent<GenderIdProps> = ({
+  genderId,
+  hint,
+  label,
+  required,
+  error,
+  value,
+  preferNotToAnswerText,
+  preferNotToAnswerChecked: initialPreferNotToAnswerChecked = false,
+}) => {
+  const { register, setValue } = useFormContext();
+  const [preferNotToAnswerChecked, setPreferNotToAnswerChecked] = useState(
+    initialPreferNotToAnswerChecked
+  );
+
+  const errorId = `input-error-message-${genderId}`;
+  const hintId = `hint-${genderId}`;
+  const preferNotToAnswerId = `prefer-not-to-answer-${genderId}`;
+
+  useEffect(() => {
```

Comment:
Is this solving the problem of making the checked value persistent between DOM updates? If so, I think using `defaultChecked` on the checkbox input (below) rather than `checked` will remove the need for the `useEffect`.